diff --git a/CMakeLists.txt b/CMakeLists.txt
index da0f4f9..efd5f42 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,13 +48,8 @@ else()
     support core irreader
     bitreader bitwriter
     passes asmprinter
-    aarch64info aarch64desc aarch64codegen aarch64asmparser
-    armcodegen armasmparser
     interpreter mcjit
-    nvptxdesc
-    x86info x86codegen x86asmparser
-    sparccodegen sparcasmparser
-    webassemblydesc)
+    x86info x86codegen x86asmparser)
 endif()
 message(STATUS "LLVM Libraries: ${llvm_libs}")
 
